src/sensors/tests/test_irradiancemeter.py

```diff
@@ -126,10 +110,9 @@ def test_incoming_flux(variant_scalar_rgb):
 
     power_density_cum = 0.0
     num_samples = 100
-    random.seed(42)
 
     for i in range(num_samples):
-        r = [random.random() for x in range(6)]
+        r = [np.random.rand() for x in range(6)]
```

Comment:
I would suggest the following:

```python
wav_samples = np.random.rand(num_samples)
pos_samples = np.random.rand(num_samples, 2)
dir_samples = np.random.rand(num_samples, 2)
for i in range(num_samples):
    ray, weight = sensor.sample_ray_differential(0.0, wav_samples[i], pos_samples[i], dir_samples[i])
```

And same above

Reply:
I suppose this is more efficient, because the RNG is calles only three times, instead of 100?
I pushed some changes.

Reply:
It is probably more efficient, but my suggestion was more about keeping the code clean 😉